nmdc_schema/migrators/migrator_from_XX_to_PR242.py

```diff
@@ -0,0 +1,32 @@
+from nmdc_schema.migrators.migrator_base import MigratorBase
+
+
+class Migrator(MigratorBase):
+    r"""Migrates a database between two schemas."""
+
+    _from_version = "XX"
+    _to_version = "XX" # PR242
+
+    def upgrade(self):
+        r"""Migrates the database from conforming to the original schema, to conforming to the new schema."""
+
+        self.adapter.process_each_document(
+            "functional_annotation_agg", [self.move_metagenome_id_to_was_generated_by]
```

Comment:
Note to self: This collection doesn't have an `id` slot. Ensure the migration framework (both the adapter methods and the standard code in the downstream migration notebooks) supports this.

Reply:
I removed the id field from the doctests as well.  This PR should enable _compound indexing_ in mongoDB on this collection (the unique_keys aspect of the PR).  I'm not sure that's related here but I wanted to keep that in the conversation since I know this collection is a bit unconventional.

Reply:
Brynn showed me how to test the migrator locally and it passed!

Reply:
Great! Thanks for confirming. Similarly, I confirmed the adapter methods and migration notebook don't depend on there being a field named `id` in the document (whew).